cmd/gctsCreateRepository.go

```diff
@@ -0,0 +1,156 @@
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+	httpClient := &piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := createRepository(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func createRepository(config *gctsCreateRepositoryOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, _ := cookiejar.New(nil)
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	type repoData struct {
+		RID             string `json:"rid"`
+		Name            string `json:"name"`
+		Role            string `json:"role"`
+		Type            string `json:"type"`
+		VSID            string `json:"vsid"`
+		GithubURLstring string `json:"url"`
+	}
+
+	type createRequestBody struct {
+		Repository string   `json:"repository"`
+		Data       repoData `json:"data"`
+	}
+
+	type repoConfig struct {
+		Key      string `json:"key"`
+		Value    string `json:"value"`
+		Category string `json:"category"`
+	}
+
+	type createResultBody struct {
+		RID         string       `json:"rid"`
+		Name        string       `json:"name"`
+		Role        string       `json:"role"`
+		Type        string       `json:"type"`
+		VSID        string       `json:"vsid"`
+		Status      string       `json:"status"`
+		Branch      string       `json:"branch"`
+		URL         string       `json:"url"`
+		CreatedBy   string       `json:"createdBy"`
+		CreatedDate string       `json:"createdDate"`
+		Connection  string       `json:"connection"`
+		Config      []repoConfig `json:"config"`
+	}
+
+	type createResponseBody struct {
+		Repository createResultBody `json:"repository"`
+		Exception  string           `json:"exception"`
+	}
+
+	reqBody := createRequestBody{
+		Repository: config.Repository,
+		Data: repoData{
+			RID:             config.Repository,
+			Name:            config.Repository,
+			Role:            config.Role,
+			Type:            config.Type,
+			VSID:            config.VSID,
+			GithubURLstring: config.RemoteRepositoryURL,
+		},
+	}
+	jsonBody, marshalErr := json.Marshal(reqBody)
+
+	if marshalErr != nil {
+		return fmt.Errorf("creating the repository locally failed: %w", marshalErr)
+	}
+
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Add("Accept", "application/json")
+
+	url := "http://" + config.Host +
```

Comment:
Is `http` always the right choise? I guess: no nowadays. I would like to suggest to have `<protocol>://<host>` rather than only `<host>` configurable either by having that in one string or by having two config properties: protocol, host.

Reply:
We only plan to support `http`. Would you still want to make any changes?

Reply:
No, in that case it is OK. Does not make sense to make something configurable where only one value exists. Wondering why http is the only supported option, but ok.

Reply:
That would be a question for @martin-bredy

Reply:
No, gCTS support both protocols http and https because of the fact that we leverage the SAP Gateway which provides the functionality.
It depends on the customer configuration of the target system which protocol is available.
Therefore, we should not enforce http here @gochax

Reply:
My mistake, I was thinking beyond `http(s)`.  Will fix.